### PR TITLE
[dev-6597] Editor reducer refactor

### PR DESCRIPTION
**Pull request by joshlacey**
This change was necessary as a preliminary for the MultiDashboard work.

### CODE REVIEW

File: packages/core/types/Visualization.ts [hunk 1]
```diff
@@ -2,20 +2,26 @@ export type Visualization = {
   autoLoad: boolean
   data: any
   dataDescription: Object
+  dataFileName: string
+  dataFileSourceType: string
   dataKey: string
+  datasets: any
```
Comment (adamdoe) [inline packages/core/types/Visualization.ts:8]: I think this should be a record.

File: packages/core/types/Visualization.ts [hunk 2]
```diff
@@ -2,20 +2,26 @@ export type Visualization = {
...
   editing: boolean
-  formattedData: any
+  formattedData?: Object[]
   general: any
   hide: any[]
+  multiDashboards?: any[]
   newViz: boolean
   openModal: boolean
   originalFormattedData: any
+  orientation: any
```
Comment (adamdoe) [inline packages/core/types/Visualization.ts:17]: Likely 'vertical' | 'horizontal'

File: packages/core/types/Visualization.ts [hunk 3]
```diff
@@ -2,20 +2,26 @@ export type Visualization = {
...
+  dataFileName: string
+  dataFileSourceType: string
   dataKey: string
+  datasets: any
   editing: boolean
-  formattedData: any
+  formattedData?: Object[]
   general: any
   hide: any[]
+  multiDashboards?: any[]
```
[the remark this answers is not in the document]
Comment (joshlacey) [reply packages/core/types/Visualization.ts:13]: I'll fix this in the next PR. I forgot to remove it, was added prematurely.

File: packages/core/helpers/withDevTools.ts [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+  send: (action: Action, newState: any) => any
+  unsubscribe: () => any
+  dispatch: (action: Action) => any
+  disconnect: () => any
+}
+
+// constants
+const withDevTools = typeof window !== 'undefined' && (window as any).__REDUX_DEVTOOLS_EXTENSION__
+const devTools: DevTools = !withDevTools ? null : (window as any).__REDUX_DEVTOOLS_EXTENSION__.connect()
+const devToolsStore = !withDevTools ? null : devTools
```
Comment (adamdoe) [inline packages/core/helpers/withDevTools.ts:42]: I think you might need to add an export here. I'm seeing the build fail without that. Can you add that in and re-verify that all actions are working? I'm seeing it working in redux dev tools now, but a second set of eyes would be nice.

Comment (joshlacey) [reply packages/core/helpers/withDevTools.ts:42]: good catch! I forgot I needed to export that. so basically as I have it designed right now it gets initialized by the Editor component. so only if you're running the editor you'll see the devtools stuff.

**Pull request merged**